go-controller/pkg/node/gateway.go

```diff
@@ -431,10 +431,14 @@ func bridgeForInterface(intfName, nodeName, physicalNetworkName string, gwIPs []
 		// gateway interface is an OVS bridge
 		uplinkName, err := getIntfName(intfName)
 		if err != nil {
-			return nil, errors.Wrapf(err, "Failed to find intfName for %s", intfName)
+			if config.Gateway.Mode == config.GatewayModeLocal {
```

Comment:
Also you need to check if a user provided us with the bridge. If  a user provided us with a config, and we cant find an uplink then I would think that is an error. 

if config.Gateway.Mode == config.GatewayModeLocal  && config.Gateway.Interface == "" {

}

Reply:
Why do we want to block such a case? That is what I plan to do for microshift. I will create a `br-ex` without an uplink and pass it with `--gateway-interface=br-ex` to ovnkube.

Reply:
This is why everything worked with a dummy interface.

OVN is only able to create a bridge for an uplink. We either support creating a bridge with no uplink but then it looks we need extra parameters to decide the bridge details (name, ip, ...) or we support a pre-created bridge with no uplink which is what is intended here.